packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -349,28 +353,38 @@ class RunCommand extends RunCommandBase {
         dartDefines: dartDefines,
       );
       AppInstance app;
-      try {
-        final String applicationBinaryPath = stringArg('use-application-binary');
-        app = await daemon.appDomain.startApp(
-          devices.first, fs.currentDirectory.path, targetFile, route,
-          _createDebuggingOptions(), hotMode,
-          applicationBinary: applicationBinaryPath == null
-              ? null
-              : fs.file(applicationBinaryPath),
-          trackWidgetCreation: boolArg('track-widget-creation'),
-          projectRootPath: stringArg('project-root'),
-          packagesFilePath: globalResults['packages'] as String,
-          dillOutputPath: stringArg('output-dill'),
-          ipv6: ipv6,
-        );
-      } catch (error) {
-        throwToolExit(error.toString());
-      }
-      final DateTime appStartedTime = systemClock.now();
-      final int result = await app.runner.waitForAppToFinish();
-      if (result != 0) {
-        throwToolExit(null, exitCode: result);
-      }
+      DateTime appStartedTime;
```

Comment:
I'd like to avoid creating nested DI scopes if possible. Could we plumb this value directly through to the daemon instead?

Reply:
I pushed be95c6399b9c854000036f5cc374b7bacda2f628 - is this the sort of thing you had in mind? The condition is now handled in daemon.dart (it passes the urlTunneller function depending on the options).

Reply:
Looks good to me, thank you!